docs/spec2006-tutorial.md

```diff
@@ -0,0 +1,517 @@
+    typ = 'git repo',
+    name = 'experiment',
+    path =  './',
+    cwd = './',
+    documentation = 'local repo to run spec with gem5'
+)
+
+```
+We use .gitignore file to ignore changes of certain files or folders. 
+In this experiment, we will use this .gitignore file,
+```
+*.pyc
+m5out
+.vscode
+results
+gem5art-env
+disk-image/packer
+disk-image/packer_1.4.5_linux_amd64.zip
+disk-image/spec2006/spec2006-image/spec2006
+disk-image/spec2006
+disk-image/packer_cache
+disk-image/spec2006/CPU2006v1.0.1.iso
+gem5
+linux-4.19.83/
+```
+Essentially, we will ignore files and folders that when we use gem5art to keep track of them, or the presence of those files and folder do not affect the experiment's results.
+### Building gem5  
+In this step, we download the source code and build gem5. 
+In this tutorial, we use m5 writefile function to copy the file from the disk image to the host system.
+That function does not work out-of-the-box in the current version of gem5 (as of November 2019). 
+We need to cherry-pick two commits, one from googlesource, and one from darchr/gem5 on GitHub for that function to work. 
+```sh
+git clone https://gem5.googlesource.com/public/gem5
+cd gem5
+git remote add darchr https://github.com/darchr/gem5
+git fetch darchr  
+git cherry-pick 6450aaa7ca9e3040fb9eecf69c51a01884ac370c  
+git cherry-pick 3403665994b55f664f4edfc9074650aaa7ddcd2c
+scons build/X86/gem5.opt -j8
+```
+We have two artifacts: one is the gem5 source code (the gem5 git repo), and the gem5 binary. 
+The documentation of this step would be how we get the source code and how we compile the gem5 binary. 
+In launch_spec_experiments.py, we document the step in Artifact objects as follows,
+```python
+gem5_repo = Artifact.registerArtifact(
+    command = '''git clone https://gem5.googlesource.com/public/gem5
+		 cd gem5
+		 git remote add darchr https://github.com/darchr/gem5
+	         git fetch darchr  
+	         git cherry-pick 6450aaa7ca9e3040fb9eecf69c51a01884ac370c  
+	         git cherry-pick 3403665994b55f664f4edfc9074650aaa7ddcd2c''',
+    typ = 'git repo',
+    name = 'gem5',
+    path =  'gem5/',
+    cwd = './',
+    documentation = 'cloned gem5 master branch from googlesource and cherry-picked 2 commits on Nov 20th'
+)
+
+gem5_binary = Artifact.registerArtifact(
+    command = 'scons build/X86/gem5.opt -j8',
+    typ = 'gem5 binary',
+    name = 'gem5',
+    cwd = 'gem5/',
+    path =  'gem5/build/X86/gem5.opt',
+    inputs = [gem5_repo,],
+    documentation = 'compiled gem5 binary right after downloading the source code'
+)
+```
+### Building m5  
+m5 is a binary that facilitates the communication between the host system and the guest system (gem5). 
+The examples of how m5 is used could be found in the runscripts that we will describe later. 
+m5 binary will be copied to the disk image, so that the guest could run m5 binary during the experiment. 
+Therefore, m5 binary should be compiled before we build the disk image.  
```

Comment:
Add "Note: it's important to compile the m5 binary with `-DM5_ADDR=0xFFFF0000` as is default in the Makefile. If you don't compile with `-DM5_ADDR` and try to run with KVM you'll get an illegal instruction error."